### PR TITLE
Fix invalid API errors normalization

### LINKED ISSUE #14391: API errors are not handled as intented in API Platform
**Sylius version affected**: 1.10.x, 1.12-alpha

**Description**  
I'm working on Sylius 1.12 and APIP 2.6 (ATM, will update soon) and I can't have **some** errors shown.
The constraints list errors works, ApiPlatform\Core\Exception\ItemNotFoundException do not work.

`api_platform.yaml`
```
exception_to_status:
    App\Exception\GenericException: 400
```
`Handler.php`
```
try {
    $attribute = $this->iriConverter->getItemFromIri($attribute);
} catch (ItemNotFoundException|InvalidArgumentException $exception) {
    throw new GenericException("Not found error TEST");
}
```

And my API response is :
```
{
    "code": 500,
    "message": "Internal Server Error"
}
```
I can't manage to have error message. The HTTP code is converted to 400.

The statements here do not works: https://api-platform.com/docs/core/errors/#message-scope

**Steps to reproduce**  
Create a project, use the API and use an invalid IRI for exemple.

**Possible Solution**  
This is a bug but a workaround has been given to me.

```
<?php

namespace App\EventListener;

use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
use Symfony\Component\HttpFoundation\JsonResponse;
use Symfony\Component\HttpKernel\Event\ExceptionEvent;

#[AsEventListener(event: 'kernel.exception', method: 'onKernelException', priority: 2)]
class ExceptionListener
{
    public function onKernelException(ExceptionEvent $event): void
    {
        $exception = $event->getThrowable();

        if (str_contains($event->getRequest()->getPathInfo(), '/api/')) {
            $response = new JsonResponse([
                'message' => $exception->getMessage(),
            ],
                $exception->getCode() === 0 ? 400 : $exception->getCode()
            );
            $event->setResponse($response);
        }
    }
}
```

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.11                 |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no  |
| Related tickets | fixes #14391                      |
| License         | MIT                                                          |

So far, on production environment when any non-validation related error occurred, the `FOS\RestBundle\Serializer\Normalizer\FlattenExceptionHandler` was used, even on API Platform endpoints. Now, depending on the path of the request, the `FOS\RestBundle\Serializer\Normalizer\FlattenExceptionHandler` or `ApiPlatform\Hydra\Serializer\ErrorNormalizer` is used.

### CODE REVIEW

File: UPGRADE-API-1.11.md [hunk 1]
```diff
@@ -392,3 +392,13 @@
 
 1. Added response body field `originalUnitPrice` to the following endpoint:
    - `'GET', '/api/v2/admin/orders/{tokenValue}'`
+
+# UPGRADE FROM `v1.11.11` TO `v1.11.12`
+
+1. So far, on production environment when any non-validation related error occurred, the `FOS\RestBundle\Serializer\Normalizer\FlattenExceptionHandler` was used, even on API Platform endpoints.
+    Now, depending on the path of the request, the `FOS\RestBundle\Serializer\Normalizer\FlattenExceptionHandler` or `ApiPlatform\Hydra\Serializer\ErrorNormalizer` is used. If your code
```
Comment (Nek-) [inline UPGRADE-API-1.11.md:399]: This impact the format of the error output, right? If that is so, an example of the change may be useful.

Comment (jakubtobiasz) [reply UPGRADE-API-1.11.md:399]: Hi @Nek-. Thank you for your suggestion. As you pointed such example snippet as useful, I've added it to the upgrade file :).

File: UPGRADE-API-1.11.md [hunk 2]
```diff
@@ -392,3 +392,32 @@
 
 1. Added response body field `originalUnitPrice` to the following endpoint:
    - `'GET', '/api/v2/admin/orders/{tokenValue}'`
+
+# UPGRADE FROM `v1.11.11` TO `v1.11.12`
```
Comment (GSadee) [inline UPGRADE-API-1.11.md:396]: You should add this section as the first one in this file

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jakub! :1st_place_medal:

**Pull request merged**